api/clients/accountant.go

```diff
@@ -0,0 +1,158 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	commonpb "github.com/Layr-Labs/eigenda/api/grpc/common"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/core/meterer"
+)
+
+type IAccountant interface {
+	AccountBlob(ctx context.Context, data []byte, quorums []uint8) (uint32, uint64, error)
+}
+
+type Accountant struct {
+	// on-chain states
+	reservation       core.ActiveReservation
+	onDemand          core.OnDemandPayment
+	reservationWindow uint32
+	pricePerSymbol    uint32
+	minNumSymbols     uint32
+
+	// local accounting
+	// contains 3 bins; circular wrapping of indices
+	binRecords        []BinRecord
+	usageLock         sync.Mutex
+	cumulativePayment *big.Int
+	stopRotation      chan struct{}
+
+	paymentSigner core.PaymentSigner
+}
+
+type BinRecord struct {
+	Index uint32
+	Usage uint64
+}
+
+func NewAccountant(reservation core.ActiveReservation, onDemand core.OnDemandPayment, reservationWindow uint32, pricePerSymbol uint32, minNumSymbols uint32, paymentSigner core.PaymentSigner) *Accountant {
```

Comment:
I think a lot of what's going on here will be replaced by https://github.com/Layr-Labs/eigenda/pull/835 . binRecords and cumulativePayment should change more frequently compared to the global parameters, and no one else needs access to the structs. would you help me understand what we will get by using pointers here?

Reply:
Passing by values will make copies of the structs. Unless you're modifying the input structs, it's more efficient to pass by pointer

Reply:
New accountant would modify the input structs in later functions. Since there's no other use for the input params after being passed in, I thought it would be good for the accountant to take ownership of the params and imagined a garbage collector. Now I realize I'm thinking in rust 😅 gonna update this to be more efficient